eZ/Publish/Core/Persistence/Legacy/Content/Handler.php

```diff
@@ -554,6 +554,11 @@ public function deleteVersion( $contentId, $versionNo )
         $this->contentGateway->deleteRelations( $contentId, $versionNo );
         $this->contentGateway->deleteVersions( $contentId, $versionNo );
         $this->contentGateway->deleteNames( $contentId, $versionNo );
+
+        if ( $this->contentGateway->getLastVersionNumber( $contentId ) == 0 )
```

Comment:
@pspanja yes, but again, a bit weird that something that a value that is incremented one by one can be 0 sometime in the future :)
but this is out of the scope of what @joaoinacio is doin here. just wanted to share my thoughts with you :)